src/Sarif.Converters/ToolFileConverterBase.cs

```diff
@@ -49,6 +49,11 @@ protected internal string AddLogicalLocation(LogicalLocation logicalLocation, st
                 ++disambiguator;
             }
 
+            if (disambiguator > 0)
+            {
+                logicalLocation.FullyQualifiedName = logicalLocationKey;
```

Comment:
Bug. You correctly determined that the FQLN (`"A.B.C"`) is different from the key (`"A.B.C-1"` or whatever). But then you need to set `logicalLocation.FullyQualifiedName` to the FQLN, _not_ to the key.

As the code stands, you'll get:
```
"logicalLocations": {
  "A.B.C-1": {
    "fullyQualifiedName": "A.B.C-1"
   }
}
```
You want:
```
"logicalLocations": {
  "A.B.C-1": {
    "fullyQualifiedName": "A.B.C"
   }
}
```
Since your tests are passing, it's clear we never had a test for this corner case. Now's the time to add one. It would show that if you add two different logical locations with the same FQLN to the `logicalLocations` dictionary, you'll get two dictionary elements with different keys but the same `fullyQualifiedName`.

Reply:
But the disambiguated string is in generatedKey. logicalLocationKey is the original unchanged value

Reply:
Oh! Then consider whether we should rename it.

Reply:
string fullyQualifiedLogicalName ... ?

Reply:
Sounds good!